src/components/WideRHPContextProvider/index.tsx

```diff
@@ -91,77 +127,189 @@ const expenseReportSelector = (reports: OnyxCollection<Report>) => {
-        }
-        const lastFullScreenRouteIndex = state?.routes.findLastIndex((route) => isFullScreenName(route.name));
-        const lastRHPRouteIndex = state?.routes.findLastIndex((route) => route.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR);
+    const syncWideRHPKeys = useCallback(() => {
+        setWideRHPRouteKeys(getCurrentWideRHPKeys(allWideRHPRouteKeys, getLastVisibleRHPRouteKey(navigationRef.getRootState())));
+    }, [allWideRHPRouteKeys]);
 
-        // Both routes have to be present and the RHP have to be after last full screen for it to be visible.
-        if (lastFullScreenRouteIndex === -1 || lastRHPRouteIndex === -1 || lastFullScreenRouteIndex > lastRHPRouteIndex) {
-            return undefined;
-        }
+    const syncSuperWideRHPKeys = useCallback(() => {
+        setSuperWideRHPRouteKeys(getCurrentWideRHPKeys(allSuperWideRHPRouteKeys, getLastVisibleRHPRouteKey(navigationRef.getRootState())));
+    }, [allSuperWideRHPRouteKeys]);
 
-        return state?.routes.at(lastRHPRouteIndex)?.key;
-    });
+    const clearWideRHPKeys = useCallback(() => {
+        setWideRHPRouteKeys([]);
+        setSuperWideRHPRouteKeys([]);
+    }, []);
 
-    const wideRHPRouteKeys = useMemo(() => {
-        const rootState = navigationRef.getRootState();
+    useEffect(() => {
+        syncWideRHPKeys();
+    }, [allWideRHPRouteKeys, syncWideRHPKeys]);
 
-        if (!rootState) {
-            return [];
-        }
+    useEffect(() => {
+        syncSuperWideRHPKeys();
+    }, [allSuperWideRHPRouteKeys, syncSuperWideRHPKeys]);
 
-        const lastRHPRoute = rootState.routes.find((route) => route.key === lastVisibleRHPRouteKey);
+    const isWideRHPFocused = useMemo(() => {
+        return !!focusedRouteKey && wideRHPRouteKeys.includes(focusedRouteKey);
+    }, [focusedRouteKey, wideRHPRouteKeys]);
 
-        if (!lastRHPRoute) {
-            return [];
-        }
+    const getIsSuperWideRHPBelowFocusedScreen = useCallback(
+        (state: NavigationState | undefined, lastVisibleRouteKey: string | undefined) => {
+            if (!state) {
+                return false;
+            }
+
+            const focusedRoute = findFocusedRoute(state);
+
+            // Shouldn't ever happen but for type safety
+            if (!focusedRoute?.key) {
+                return false;
+            }
+
+            const currentSuperWideRHPRouteKeys = getCurrentWideRHPKeys(allSuperWideRHPRouteKeys, lastVisibleRouteKey);
+            const isFocusedRouteSuperWide = isSuperWideRHPRouteName(focusedRoute.name);
 
-        const lastRHPKeys = extractNavigationKeys(lastRHPRoute.state);
-        const currentKeys = allWideRHPRouteKeys.filter((key) => lastRHPKeys.has(key));
+            return currentSuperWideRHPRouteKeys.length > 0 && !currentSuperWideRHPRouteKeys.includes(focusedRoute.key) && !isFocusedRouteSuperWide;
+        },
+        [allSuperWideRHPRouteKeys],
+    );
+
+    const getIsWideRHPBelowFocusedScreen = useCallback(
```

Comment:
can you please add a comment here explaining what the value represents/ what it is for in plain english